### PR TITLE
WIP: Add initial web ui

### PR DESCRIPTION
**Pull request by david-martin**
* Generate an AngularJS app using yeoman generator-angular
* Add patternfly & origin-web-common bower deps
* Init origin-web-common from a config.js with defaults for local dev
* Add sample serviceaccount for oauthclient per namespace
* Add oauth & error pages for handling oauth redirect based login
* Add mock mobileapps & mobileapp route
* Serve all static content from "/console", with a fallback to "index.html" to allow AngularJS routes to work with html5mode

TODO:
- [ ] update origin-web-common to include scope change https://github.com/openshift/origin-web-common/pull/151
- [x] serve web content from golang server
- [x] disable auth for some endpoints (e.g. oauth/token, web content)
- [ ] allow config of oauth client id, secret & urls from a central config for both the server & the UI

### CODE REVIEW

**Comment by david-martin** [comment]
@maleck13 could you help with a few of the server side items in the todo list before we can merge this?

File: pkg/web/mobileappHandler.go [hunk 1]
```diff
@@ -16,6 +16,18 @@ type MobileAppHandler struct {
 	tokenClientBuilder mobile.TokenScopedClientBuilder
 }
 
+// OauthHandler handle oauth actions
```
Comment (maleck13) [inline pkg/web/mobileappHandler.go]: I think we should move the oauth handler to its own handler under the web directory.

File: pkg/web/routes.go [hunk 2]
```diff
@@ -12,13 +18,31 @@ import (
 	"github.com/pkg/errors"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"golang.org/x/oauth2"
 	kerror "k8s.io/apimachinery/pkg/api/errors"
 )
 
+var (
```
Comment (maleck13) [inline pkg/web/routes.go]: I think maybe we can move this to the OauthRoute function below

File: pkg/web/routes.go [hunk 3]
```diff
@@ -12,13 +18,31 @@ import (
...
+		Scopes:       []string{"user:info user:check-access"},
+		Endpoint:     *kubernetesOauthEndpoint,
+	}
+)
+
 // NewRouter sets up the HTTP Router
 func NewRouter() *mux.Router {
 	r := mux.NewRouter()
 	r.StrictSlash(true)
+	oauth2.RegisterBrokenAuthHeaderProvider(kubernetesOauthEndpoint.TokenURL)
```
Comment (maleck13) [inline pkg/web/routes.go]: same here can move down to the setup function to keep it all together

File: pkg/web/routes.go [hunk 4]
```diff
@@ -40,6 +64,108 @@ func BuildHTTPHandler(r *mux.Router, access *middleware.Access) http.Handler {
 	return n
 }
 
+func handleOauthToken(w http.ResponseWriter, r *http.Request) {
```
Comment (maleck13) [inline pkg/web/routes.go]: move into the oauth handler

File: pkg/web/routes.go [hunk 5]
```diff
@@ -40,6 +64,108 @@ func BuildHTTPHandler(r *mux.Router, access *middleware.Access) http.Handler {
...
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(tokenJSON)
+
+}
+
+// StaticRoute configures & sets up the /console route.
+func StaticRoute(r *mux.Router, staticDirectory *string) {
```
Comment (maleck13) [inline pkg/web/routes.go]: move into static handler

**Comment by david-martin** [comment]
@maleck13 I've address all comments based on your suggestions.
The only thing I think I did differently than you suggested was moving the oauth config up to main.go.
I then re-used the sa token config for the oauth client secret.

Would you prefer it to be abstracted down into the OAuthHandler and add the config bits of config to the OAuthHandler constructor?

**Comment by david-martin** [comment]
follow up issues #9 & #10 created

**Pull request merged**